LEAF_Request_Portal/form.php

```diff
@@ -1187,159 +1187,156 @@ public function doModify($recordID)
-        }
+            if (!$hasInitialStep) {
+                $return_value = array('status' => 1, 'errors' => array('Workflow is configured incorrectly'));
+            } else {
+                $vars = array(':recordID' => $recordID,
+                            ':time' => time(), );
+                $res = $this->db->prepared_query('UPDATE records SET
+                                                    submitted=:time,
+                                                    isWritableUser=0,
+                                                    lastStatus = "Submitted"
+                                                    WHERE recordID=:recordID', $vars);
 
-        // mark form as submitted, dependencyID 5 = submitted form
-        $vars = array(':recordID' => $recordID,
-                      ':dependencyID' => 5,
-                      ':filled' => 1,
-                      ':time' => time(), );
-        $res = $this->db->prepared_query('INSERT INTO records_dependencies (recordID, dependencyID, filled, time)
-                                            VALUES (:recordID, :dependencyID, :filled, :time)
-                                            ON DUPLICATE KEY UPDATE filled=:filled, time=:time', $vars);
-
-        $this->db->commitTransaction();
-
-        $errors = array();
-        // trigger initial submit event
-        include_once 'FormWorkflow.php';
-        $FormWorkflow = new FormWorkflow($this->db, $this->login, $recordID);
-        $FormWorkflow->setEventFolder('../scripts/events/');
-        foreach ($workflowIDs as $id)
-        {
-            // The initial step for Requestor is special step id -1
-            $status = $FormWorkflow->handleEvents($id, -1, 'submit', '');
-            if (count($status['errors']) > 0)
-            {
-                try {
-                    $errors = array_merge($errors, $status['errors']);
-                } catch (TypeError $te) {
-                    error_log($te);
+                // write history data, actionID 6 = filled dependency
+                $vars = array(':recordID' => $recordID,
+                            ':userID' => $this->login->getUserID(),
+                            ':dependencyID' => 5,
+                            ':actionType' => 'submit',
+                            ':actionTypeID' => 6,
+                            ':time' => time(),
+                            ':comment' => '', );
+                $res = $this->db->prepared_query('INSERT INTO action_history (recordID, userID, dependencyID, actionType, actionTypeID, time, comment)
+                                                    VALUES (:recordID, :userID, :dependencyID, :actionType, :actionTypeID, :time, :comment)', $vars);
+
+                // populate dependency data using new workflow system
+                $vars = array(':recordID' => $recordID);
+                $res = $this->db->prepared_query('SELECT * FROM category_count
```

Comment:
select *?

Reply:
This particular method I cleaned up in that I changed it so that there is only one return. I didn't really change the content, only the way it was laid out. If you'd like I can go ahead and make these changes as well.